nvflare/client/in_process/api.py

```diff
@@ -0,0 +1,206 @@
+        if self.__continue_job():
+            self.logger.info("send local model back to peer ")
+
+        if self.client_config.get_transfer_type() == "DIFF":
+            model = self._prepare_param_diff(model)
+
+        shareable = FLModelUtils.to_shareable(model)
+        self.event_manager.fire_event(TOPIC_LOCAL_RESULT, shareable)
+
+        if clear_cache:
+            self.fl_model = None
+
+    def system_info(self) -> Dict:
+        return self.sys_info
+
+    def get_config(self) -> Dict:
+        return self.client_config.get_config()
+
+    def get_job_id(self) -> str:
+        return self.meta[FLMetaKey.JOB_ID]
+
+    def get_site_name(self) -> str:
+        return self.meta[FLMetaKey.SITE_NAME]
+
+    def get_task_name(self) -> str:
+        return self.meta[ConfigKey.TASK_NAME]
+
+    def is_running(self) -> bool:
+        if not self.__continue_job():
+            return False
+        else:
+            self.receive()
+
+        if self.fl_model:
+            self.current_round = self.fl_model.current_round
+            self.total_rounds = self.fl_model.total_rounds
+            self.start_round = self.fl_model.meta.get(FLMetaKey.START_ROUND, 0)
+        else:
+            return False
+
+        return self.current_round < self.start_round + self.total_rounds
+
+    def is_train(self) -> bool:
+        return self.meta.get(ConfigKey.TASK_NAME) == self.client_config.get_train_task()
+
+    def is_evaluate(self) -> bool:
+        return self.meta.get(ConfigKey.TASK_NAME) == self.client_config.get_eval_task()
+
+    def is_submit_model(self) -> bool:
+        return self.meta.get(ConfigKey.TASK_NAME) == self.client_config.get_submit_model_task()
+
+    def log(self, key: str, value: Any, data_type: AnalyticsDataType, **kwargs):
+        msg = dict(key=key, value=value, data_type=data_type, **kwargs)
+        self.event_manager.fire_event(TOPIC_LOG_DATA, msg)
+
+    def clear(self):
+        self.fl_model = None
+
+    def _prepare_param_diff(self, model: FLModel) -> FLModel:
+        exchange_format = self.client_config.get_exchange_format()
+        diff_func = DIFF_FUNCS.get(exchange_format, None)
+
+        if diff_func is None:
+            raise RuntimeError(f"no default params diff function for {exchange_format}")
+        elif self.fl_model is None:
+            raise RuntimeError("no received model")
+        elif self.fl_model.params is not None:
+            if model.params_type == ParamsType.FULL:
+                try:
+                    model.params = diff_func(original=self.fl_model.params, new=model.params)
+                    model.params_type = ParamsType.DIFF
+                except Exception as e:
+                    raise RuntimeError(f"params diff function failed: {e}")
+
+        if model.params is None and model.metrics is None:
+            raise RuntimeError("the model to send does not have either params or metrics")
+
+        return model
```

Comment:
some duplicated codes in `ModelRegistry` as well, we need to re-factor them to a common function, it can be done later, just a note.